File: src/main/java/org/kite9/diagram/visualization/planarization/rhd/grouping/GroupingStrategy.java
package org.kite9.diagram.visualization.planarization.rhd.grouping;

import org.kite9.diagram.visualization.planarization.rhd.GroupBuilder;
import org.kite9.diagram.visualization.planarization.rhd.GroupPhase;

/**
 * Defines a way of merging leaf groups within the {@link GroupPhase} together to create a single compound group.
 * 
 * @author robmoffat
 *
 */
public interface GroupingStrategy extends GroupBuilder {

	public GroupResult group(GroupPhase gp);
			
}

File: src/main/kotlin/org/kite9/diagram/visualization/planarization/rhd/grouping/GroupingStrategy.kt
package org.kite9.diagram.visualization.planarization.rhd.grouping

import org.kite9.diagram.visualization.planarization.rhd.GroupBuilder
import org.kite9.diagram.visualization.planarization.rhd.GroupPhase

/**
 * Defines a way of merging leaf groups within the [GroupPhase] together to create a single compound group.
 *
 * @author robmoffat
 */
interface GroupingStrategy {

    fun group(): GroupResult
}